File: Proyecto-PracticasProfesionales/src/domain/empresa/Empresa.ts
export interface Empresa {
  id_empresa: number;
  id_usuario: number | null;
  nombre_empresa: string;
  rfc: string | null;
  giro: string | null;
  domicilio: string | null;
  telefono: string | null;
  correo_contacto: string | null;
  estado_empresa: string;
}

export interface CrearEmpresaDTO {
  id_usuario?: number | null;
  nombre_empresa: string;
  rfc?: string | null;
  giro?: string | null;
  domicilio?: string | null;
  telefono?: string | null;
  correo_contacto?: string | null;
}

File: Proyecto-PracticasProfesionales/backend/models/empresa.py
from sqlalchemy import Column, Integer, String, Text, Enum
from database.connection import Base


class EmpresaModel(Base):
    __tablename__ = "empresa"

    id_empresa = Column(
        Integer,
        primary_key=True,
        autoincrement=True
    )

    id_usuario = Column(
        Integer,
        nullable=True
    )

    nombre_empresa = Column(
        String(150),
        nullable=False
    )

    rfc = Column(
        String(20),
        nullable=True
    )

    giro = Column(
        String(100),
        nullable=True
    )

    domicilio = Column(
        Text,
        nullable=True
    )

    telefono = Column(
        String(15),
        nullable=True
    )

    correo_contacto = Column(
        String(100),
        nullable=True
    )

    estado_empresa = Column(
        Enum("Pendiente", "Activa", "Suspendida", "Inactiva"),
        default="Pendiente"
    )